docs/semantics/encapsulation.md

```diff
@@ -74,10 +74,14 @@ public and private constructors in a single type is a compilation error. A type
 with all constructors public is called an _open_ type and a type with all
 constructors private is called a _closed_ type.
 
+### Methods
+
 Methods on types (or on modules) can be specified private. To check whether a
 private method is accessed only from within the same project, a runtime check
 must be performed, as this cannot be checked during the compilation.
 
+Conversion and foreign methods cannot be specified as private.
```

Comment:
There probably is no (technical) reason why a `foreign` method couldn't be `private`, I think.

Reply:
Correct. Other than probably non-trivial changes in the parser once again. The following snippet:
```
private foreign js js_func = """
    return 42; 
```
parses to:
```
(Private (Function (Ident foreign) #((() (Ident js) () ()) (() (Ident js_func) () ())) () "=" (TextLiteral #((Section "return 42;")))))
```
and I don't know (and don't want to deal with it in this PR) how to convince the parser to parse it differently.

Let's skip the private declaration for foreign methods for now. If there is a use-case for them in the future, let's create another ticket for that.

Reply:
Created a follow-up https://github.com/enso-org/enso/issues/10137 for this. Let's not solve private foreign functions in this PR.